src/libDirectoryService/DSBlockPreProcessing.cpp

```diff
@@ -1071,6 +1217,20 @@ bool DirectoryService::DSBlockValidator(
     return false;
   }
 
+  // Check if the current block version to be validated requires removed nodes
+  // validation.
+  const uint32_t REMOVED_FIELD_DSBLOCK_VERSION = 2;
+  if (m_pendingDSBlock->GetHeader().GetVersion() >=
+          REMOVED_FIELD_DSBLOCK_VERSION &&
+      m_mediator.m_currentEpochNum >= ENABLE_NEW_FEATURE_DS_NUM) {
```

Comment:
Change to m_mediator.m_dsBlockChain.GetLastBlock().GetHeader().GetBlockNum() >=
      ENABLE_NEW_FEATURE_DS_NUM